pkg/apis/certificates/validation/validation.go

```diff
@@ -58,6 +60,54 @@ func ValidateCertificateSigningRequest(csr *certificates.CertificateSigningReque
 	if len(csr.Spec.Usages) == 0 {
 		allErrs = append(allErrs, field.Required(specPath.Child("usages"), "usages must be provided"))
 	}
+	if csr.Spec.SignerName == nil {
+		allErrs = append(allErrs, field.Required(specPath.Child("signerName"), "signerName must be provided"))
+	} else {
+		// ensure signerName is of the form domain.com/something and up to 571 characters.
+		// This length and format is specified to accommodate signerNames like:
+		// <resource-type-name>.<fully-qualified-group-name>/<resource-namespace>.<resource-name>
+		if len(*csr.Spec.SignerName) > 571 {
```

Comment:
Thanks for forcing this one.. because after I started adding things up, I got to 635 🙃 

```
		// ensure signerName is of the form domain.com/something and up to 635 characters.
		// This length and format is specified to accommodate signerNames like:
		// <resource-type-name>.<fully-qualified-group-name>/<resource-namespace>.<resource-name>.
		// The max length of a resource type name is 63 characters (DNS1035Label max length)
		// The max length of a group name is 253 characters (DNS1123Subdomain max length)
		// The max length of a namespace name is 63 characters (DNS1123Label max length)
		// The max length of a resource name is 253 characters (DNS1123Subdomain max length)
		// We then add an additional 3 characters to account for the two '.' and one '/'.
```

Does that make sense?

Reply:
cc @liggitt as you initially came up with 571 and this is what is in the KEP